### ISSUE TITLE
CrudAuth persist function not adding property to DTO

### ISSUE THREAD

**Issue by bestickley** [nestjsx/crud#550]
I have the below persist function that I want to add a property to the DTO but it's not working. Does anyone know why? I tried to find the code that merges the returned object from the persist function into the DTO but could not. Am I misinterpreting the docs?
```
persist: (req: Request) => {
    if (req.session) {
      if (req.method === "POST") {
        return { createdBy: { id: req.session.user } };
      } else if (req.method === "PATCH" || req.method === "PUT") {
        return { updatedBy: { id: req.session.user } };
      } else {
        return {};
      }
    } else {
      return {};
    }
  }
```

> `persist` - a function that can return an object that will be added to the DTO on create, update

**Comment by zMotivat0r** [comment]
are you sure you have `req.session` defined?

**Comment by bestickley** [comment]
@zMotivat0r, thank you for your quick response. Yes, I'm sure req.session is defined.
Answered here: #506
tl;dr: the property is added at the final TypeORM save function call, not in the dto object within the controller.

**Comment by bestickley** [comment]
Never mind. That actually didn't work. It would work if I had `returnShallow` to be true but when it's false (by default) it doesn't grab `updatedBy`

**Comment by bestickley** [comment]
I found the issue:
```
async updateOne(req, dto) {
        const { allowParamsOverride, returnShallow } = req.options.routes.updateOneBase;
        const paramsFilters = this.getParamFilters(req.parsed);
        const found = await this.getOneOrFail(req, returnShallow);
        const toSave = !allowParamsOverride
            ? { ...found, ...dto, ...paramsFilters, ...req.parsed.authPersist }
            : { ...found, ...dto, ...req.parsed.authPersist };
        const updated = await this.repo.save(class_transformer_1.plainToClass(this.entityType, toSave));
        if (returnShallow) {
            return updated;
        }
        else {
            req.parsed.paramsFilter.forEach((filter) => {
                filter.value = updated[filter.field];
            });
            return this.getOneOrFail(req);
        }
    }
```
Both properties I was trying to return (createdBy and updatedBy) are relations to my User entity. Therefore, in the last line (`return this.getOneOrFail(req);`) it doesn't join the User table in the query and it's not pulled. I need to add updatedById and createdById are on the User table. I've been using `@JoinColumn({ name: "updatedBy" })` and `@JoinColumn({ name: "createdBy" })` which takes away the `columnNameId` column.
Issue solved. Hopefully this helps someone else!

**Comment by muhammadsamir11** [comment]
I have the same issue can you explain more how you fix it

**Comment by bestickley** [comment]
@muhammadsamir11, sure! The main issue I had was that I was adding a _relation_ to the DTO in my `persist` function and because @nestjsx/crud-typeorm's updateOne/createOne function use a separate `getOneOrFail` function to return the entity you just updated/created. However, the relation is added to the DB, it's just not returned by the function unless you join it.

I'm not sure why your "userId" has null values. If it's a property on your entity the `getOneOrFail` function should return it. I would debug the function to explore the issue more.
